src/variety/turnaround.js

```diff
@@ -0,0 +1,175 @@
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["turnaround"], {
+	MouseEvent: {
+		inputModes: {
+			edit: ["number", "clear", "info-line"],
+			play: ["line", "peke", "clear", "info-line"]
+		},
+		autoedit_func: "qnum",
+		autoplay_func: "line"
+	},
+	KeyEvent: {
+		enablemake: true
+	},
+
+	Cell: {
+		minnum: 0,
+		maxnum: 3,
+
+		isCmp: function() {
+			if (!this.puzzle.execConfig("autocmp")) {
+				return false;
+			}
+			return this.qnum === this.countTurn();
+		},
+
+		countTurn: function() {
+			if (!this.isNum() || this.lcnt !== 2) {
+				return -1;
+			}
+
+			var clist = this.getAdjCells();
+			if (
+				clist.some(function(cell) {
+					return cell.lcnt !== 2;
+				})
+			) {
+				return -1;
+			}
+
+			clist.add(this);
+			var turncount = 0;
+			clist.each(function(cell) {
+				turncount += cell.isLineCurve() ? 1 : 0;
+			});
+
+			return turncount;
+		},
+
+		getAdjCells: function() {
+			var clist = new this.klass.PieceList();
+			for (var dir in this.adjborder) {
+				if (this.adjborder[dir].isLine()) {
+					clist.add(this.adjacent[dir]);
+				}
+			}
+			return clist;
+		},
+
+		posthook: {
+			qans: function(num) {
```

Comment:
```suggestion
			qnum: function(num) {
```

`qans` isn't used in this puzzle, but a `qnum` hook could be useful when editing a puzzle.